### PR TITLE
Create flake.nix

### LINKED ISSUE #63: `"` crashes the shell (sometimes)

### PR DESCRIPTION
**Pull request by PointerDilemma**
I noticed that this `flake.nix` file was missing in you project so I decided to help you and create it!1111

Pls merge thx!

Fixes #63

### CODE REVIEW

**Comment by PointerDilemma** [comment]
good first issue!!!!!

**Comment by PointerDilemma** [comment]
Those changes make sesnse

**Comment by PointerDilemma** [comment]
hacktoberfest is not over yet :eyes: 

Here is a copilot summary :)))
This pull request introduces a new `flake.nix` file to set up a Nix flake for building and testing a "Hello World" program in C. The most important changes include defining the flake's structure, specifying system types, and setting up build and test configurations.

### Nix flake setup:

* [`flake.nix`](diffhunk://#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R1-R116): Added a description and specified the Nixpkgs version to use.
* [`flake.nix`](diffhunk://#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R1-R116): Defined supported system types (`x86_64-linux`, `x86_64-darwin`, `aarch64-linux`, `aarch64-darwin`).
* [`flake.nix`](diffhunk://#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R1-R116): Created a Nixpkgs overlay to define the `hello` package using `stdenv.mkDerivation`.

### Build and test configurations:

* [`flake.nix`](diffhunk://#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R1-R116): Configured the default package for `nix build` and provided binary packages for selected system types.
* [`flake.nix`](diffhunk://#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R1-R116): Set up NixOS module and tests, including integration tests and a VM test for the `hello` package.

**Comment by PointerDilemma** [comment]
I think @ecomaikgolf  does not like nix :(

**Comment by PointerDilemma** [comment]
The nix syntax is just :sweat_drops:

**Comment by PointerDilemma** [comment]
A function in nix is declared like this:

```nix
double = x: x*2
```

you are welcome

**Comment by PointerDilemma** [comment]
mate in 1

**Comment by sebastian-felix** [comment]
I think a qemu dependency is still missing, otherwise LGTM

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      # package.
+      defaultPackage = forAllSystems (system: self.packages.${system}.hello);
+
+      # A NixOS module, if applicable (e.g. if the package provides a system service).
+      nixosModules.hello =
+        { pkgs, ... }:
+        {
+          nixpkgs.overlays = [ self.overlay ];
+
+          environment.systemPackages = [ pkgs.hello ];
```
Comment (sebastian-felix) [inline flake.nix:60]: qemu

**Review by sebastian-felix — CHANGES_REQUESTED** [review]
LGTM now

**Comment by ecomaikgolf** [comment]
this is what nix does to you I guess

**Pull request closed without merging**